### DISCUSSION TITLE
Uncertain how to start a new g7 after the old expires; it just keeps saying "sensor expired"

### DISCUSSION THREAD

**Discussion by bkuz114** [NightscoutFoundation/xDrip#3730] [General]
Hello. I have successfully used xDrip with my previous Dexcom g7. I set it up using [these instructions](https://navid200.github.io/xDrip/docs/Dexcom/G7.html)

That sensor expired and it's been thrown away. I put on a new sensor and started it via my Dexcom receiver. However, it's unclear to me how to get it to work in xDrip, because xDrip keeps saying "sensor expired", as though it thinks I'm still using the previous one.

I had a look at [this page](https://navid200.github.io/xDrip/docs/Dexcom/SubsequentG7.html) but I am having a difficult time following it. Maybe it's a language issue. I don't have a second phone ('test phone') to set things up on. I did disconnect via settings -> hardware data source -> disable collection. But when I switch this back to g7 and update the pairing code to the new pairing code, it just continually says "sensor expired" (as though it thinks this is the old sensor).

Do I need to redo the entire setup again? (uninstall xdrip and start over from the beginning?)

Note that the "start a subsequent g7" instructions page, points to [this page](https://navid200.github.io/xDrip/docs/Dexcom/G7.html) again, so I did re-scan that QR code and entered the new pairing code again. On the dex status page, it does prompt me to pair with the new g7. However, the "sensor status" still says "sensor expired"

**Comment by Navid200** [comment]
We are still improving G7 functionality as much as possible.

If you use a receiver or a pump, things get a little bit tricky and I need to update the guide to cover that.

But, the first problem you are experiencing is because you just threw away the old sensor.  But, very likely, it is still close to you and not inside a metallic container.
That's why xDrip is still connected to it.  Yes, you have entered the pairing code of the new device.  This is one of the limitations of xDrip.  At least, for now.  Possibly always.
So, you should not just throw away the sensor.  You should put it inside a metallic box or if you throw it away, it should be really far AWAY.

If you are really away from the old sensor, you need to turn off your receiver.  Then, wait 15 minutes.  Now, G7 is in rapid connect mode.
Now, go to the status page and you should get a pair request.
After you do and approve it and get readings, you can restart the receiver and the two will coexist after that.

**Reply by bkuz114** [reply]
Thanks kindly for responding to this. I am really grateful for the g7 support. xdrip is life saving for me because it allows me to set the proper alarms to wake me up in my sleep from hypoglycemia events (the dexcom app won't work for my purposes).

About the old sensor: the old g7 sensor is in a garbage can probably 20 feet away from me. Is that far enough? Unfortunately I don't have the ability to move it further away, nor do I have a way to get it out of the trash. Does this mean I won't be able to use xdrip with my new sensor? Or should I just uninstall xdrip and start over again?

**Reply by Navid200** [reply]
There are so many ways to address this.  But, I am not aware of all your limitations.  Can you tell me if you can possibly move temporarily, may be for a cup of coffee or tea, to a place about 100 meters (330 feet) away from the old sensor.
As soon as xDrip gets going with the new device, you will be able to go back to close proximity of the old device with no problem.

**Reply by bkuz114** [reply]
Thanks. I can't do this right now as I have some mobility limitations. However, I can probably do this tomorrow.

Once I am away from it (and the receiver) - if I understand properly from your last post, I'll want to wait 15 minutes before I try connecting to the new device. However, I'd like to clarity what are the steps required in between: Should I forget all previous bluetooth devices again? Should I "disconnect" again and then reset the hardware data source to g7 again, or should things just work without additional steps? (That way I can try and prepare.)

**Reply by Navid200** [reply]
Is your phone compatible with the Dexcom app?

**Reply by bkuz114** [reply]
Yes it is.

**Reply by Navid200** [reply]
One option you have that does not require you to move away is this:
Change the hardware data source in xDrip from `G6/G7/1/1+ Transmitter` to `Companion App`.
Then, install Dexcom app and follow its instructions to get it going with your current G7.  

Then, follow this to get readings from Dexcom app to xDrip:
https://navid200.github.io/xDrip/docs/Follow/CompanionApp.html  
<br/>  
  
---  
  
When the time comes that you want to use xDrip directly and you are away from the old sensor, you can do the following:
1- Delete all devices whose name start with DXCM from the list of connected or previously connected devices.
2- Set Hardware data source to `G6/G7/1/1+ Transmitter`.
3- You should get a pair request.  Approve it.  It should provide readings after that.

**Reply by bkuz114** [reply]
Unfortunately I can't keep the dexcom app on the same phone as xdrip. (For context, i get frequent, severe lows throughout the day; the urgent low alarm on dexcom app - which is 55 mg/dl - can't be toggled off, so it goes off frequently in my sleep and I end up having to turn the phone off. In reality, my urgent low is more around 40 mg/dl. xDrip allows me to set such an alarm, so that I"m actually woken up when I need to be. The reason I switched from libre to g7 is because xdrip supports it, thus, I can run the phone with xdrip only, and don't have to keep the manufacturer app on my phone, which allows me to have the alarm I need.)

I will go ahead and wait until I'm outside the home, and then follow these instructions you've provided. I really appreciate it. Will update here to let know if it was successful (in case that is helpful to others who might stumble on this.)

**Reply by bkuz114** [reply]
Hey, not sure if this is useful info. But unfortunately I needed to get this working before sleep, so I went ahead and uninstalled xdrip, then re-installed xdrip and went through the set up again, and I was able to get things working.

I've made notes for myself about how to do the setup (using your instructions [here](https://navid200.github.io/xDrip/docs/Dexcom/G7.html), and just taking what's relevant for my phone), so that it's only a few concise steps. This means the set up only takes a few minutes. Going forward I might just take this approach going forward.

**Reply by Navid200** [reply]
Great

Make sure to have a certain plan for the used devices.  I put all mine inside a metallic box.

**Reply by bkuz114** [reply]
Thanks a lot, and thanks for all your wonderful work on getting the g7s to work natively with xdrip. It's lifechanging for myself, and for many others I'm certain.

I'm curious about something, if you don't mind. (BTW - if the answer is very involved or would take a lot of your time, please feel free to disregard. This is just curiosity on my end, and doesn't impact my ability to use xDrip.) How is bluetooth connection so strong on these devices (i.e. that it keeps detecting old devices), when it disconnects so easily while in use? For example, if I am just in another room, xdrip tends to disconnect and it takes 5-10 minutes for it to reconnect.

Even if I weren't using xdrip and were just using dexcom app, I feel like if I went very far away, the bluetooth connection would disconnect. It's just shocking to me that it is able to detect the old devices so strongly.

**Reply by Navid200** [reply]
You said the old device was only 20 feet away.  That is only 6 meters.  That's not too far.  

G7 changed dramatically with respect to G6.  With G7, we don't know what the Bluetooth name is just looking at the pairing code.  With G6, xDrip could figure out what the Bluetooth name would be just by having the serial number.
This has put xDrip at a disadvantage with respect to establishing a p[airing with a new G7 device.
That is why xDrip cannot distinguish between the old and the new device as easily as it could with G6.  So, we advise that you keep the old device isolated to make things easier for xDrip.

It will eventually connect as you experienced.  But, it is not 100% predictable with an old device sitting nearby.

After there is a pairing, xDrip and G7 are both looking for each other every 5 minutes.  After xDrip connects, it does what it can to stay connected.  If xDrip is connected to the old G7, it will try to stay connected with it even if you change the pairing code because of the reason I explained.  

My hope is that we can improve this.

**Reply by bkuz114** [reply]
Thanks for taking your time to explain this. That makes a lot of sense, and also explains why it keeps pairing with the old receiver even when you put a new pairing code in.

I think you guys have done a fantastic job with g7 support. Yes, the sensor disconnects frequently (as I mentioned above), but my assumption is that has nothing to do with xdrip, and everything to do with bluetooth on my phone (phone bluetooth has issues no matter what the device - lots of disconnects from say a bluetooth headphone, etc.) Aside from that it works flawlessly. Understanding now the roadblocks you guys face with g7 (that you don't know the name just from the pairing code), I'm even more impressed by how well xdrip works.
